org.mwc.debrief.legacy/src/Debrief/Wrappers/Measurements/CoreDataset.java

```diff
@@ -0,0 +1,89 @@
+package Debrief.Wrappers.Measurements;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import junit.framework.TestCase;
+import MWC.GUI.CanvasType;
+import MWC.GUI.PlainWrapper;
+import MWC.GenericData.WorldArea;
+
+public class CoreDataset<IndexType extends Number, ValueType> extends PlainWrapper implements DataItem
+{
+  /**
+   * 
+   */
+  private static final long serialVersionUID = 1L;
+  
+  private final List<Measurement> _data = new ArrayList<Measurement>();
+  private String _name;
+  
+  
+  private class Measurement
+  {
+    private IndexType _index;
+    private ValueType _value;
+    
+    @SuppressWarnings("unused")
+    public Measurement(final IndexType index, final ValueType value)
+    {
+      _index = index;
+      _value = value;
+    }
+  }
+  
+  public CoreDataset(String name)
+  {
+    _name = name;
+  }
+  
+  public void add(Measurement measurement)
+  {
+    _data.add(measurement);
+  }
+  
+  @Override
+  public String getName()
+  {
+    return _name;
+  }
+
+  @Override
+  public void paint(CanvasType dest)
+  {
+  }
+
+  @Override
+  public boolean hasEditor()
+  {
+    return false;
+  }
+
+  @Override
+  public WorldArea getBounds()
+  {
+    return null;
+  }
+  
+  public class TestCore extends TestCase
+  {
+    public void testMe()
+    {
+      CoreDataset<Long, Double> timeD = new CoreDataset<Long, Double>("TimeDouble");
+      timeD.printAll();
+      
+      // TODO:  I wish to create an instance of the measurement object - but I get compiler warning
```

Comment:
@theanuradha - I can't declare the Measurement object in the next line.  It's down to Generics...